lib/engine/step/tokener.rb

```diff
@@ -125,20 +126,17 @@ def adjust_token_price_ability!(entity, token, hex, city, special_ability: nil)
           next if ability.hexes.any? && !ability.hexes.include?(hex.id)
           next if ability.city && ability.city != city.index
 
-          # check if this is correct or should be a corporation
-          if ability.extra
-            if ability.neutral
-              neutral_corp = Corporation.new(
-                sym: 'N',
-                name: 'Neutral',
-                logo: 'open_city',
-                tokens: [0],
-              )
-              token = Engine::Token.new(neutral_corp, type: :neutral)
-            else
-              token = Engine::Token.new(entity)
-              entity.tokens << token
-            end
+          if ability.neutral
+            neutral_corp = Corporation.new(
+              sym: 'N',
+              name: 'Neutral',
+              logo: 'open_city',
+              tokens: [0],
+            )
+            token = Engine::Token.new(neutral_corp, type: :neutral)
+          elsif ability.owner.company? && !ability.from_owner
```

Comment:
This is the only real logic change in this PR.

1882 had a config change for one private - https://github.com/tobymao/18xx/pull/4932/files#diff-f3da9b37ea0b71dfa9bf440659066596d6d31095abec184956ec760b361473e3R294

A test for the DC&H company was added, and the LAT test was changed so that they could use the same fixture - https://github.com/tobymao/18xx/pull/4932/files#diff-04a1776aafdbd418b15bf5daac0af05a9f64cee365f0cfd8a4a00d8c751a8f26

All other Ruby changes in this PR are about renaming `extra` to `extra_action`